frontend/frontend-filters-links.php

```diff
@@ -403,6 +403,19 @@ public function check_canonical_url( $requested_url = '', $do_redirect = true )
 					} else {
 						$redirect_url = $this->maybe_add_page_to_redirect_url( get_term_link( $term_id ) );
 					}
+					// Keeps query string from the requested URL if any.
+					$query_string = wp_parse_url( $requested_url, PHP_URL_QUERY );
+
+					if ( is_string( $query_string ) && ! empty( $query_string ) ) {
+						// Removes query string parameters which can be used for rewriting.
+						parse_str( $query_string, $parameters );
+						foreach ( $parameters as $parameter_name => $value ) {
+							if ( in_array( $parameter_name, array( 'cat', 'tag', 'category_name', 'feed', 'paged' ), true ) ) {
+								unset( $parameters[ $parameter_name ] );
+							}
+						}
+						$redirect_url = $redirect_url . ( empty( $parameters ) ? '' : '?' . http_build_query( $parameters ) );
+					}
```

Comment:
`check_canonical_url()` is going bigger and bigger. Maybe we could move this code inside a dedicated method ?

Reply:
I asked myself this question 😁

Reply:
Done in 3cbef55b6c8ba1b15c1fb52af29425839dc5bec1